### PR TITLE
[WIP] Fix labeler action

### LINKED ISSUE #195: Setup pull request auto-labeling
Use something like this, with labels for erb/js/css, models, db/, and Pundit policy files
https://github.com/actions/starter-workflows/blob/master/automation/label.yml
https://github.com/actions/starter-workflows/blob/master/automation/greetings.yml

### PR DESCRIPTION
**Pull request by tmr08c**
### What github issue is this PR for, if any?

Resolves #195 (and fixes the version introduced in #1001 and #1006).

### What changed, and why?

* Adds workflow file for Action that uses the labeler action
* Creates `labeler.yml` file that maps labels to file changes that will
      trigger adding that label

Possibly more interesting is what is different between this and #1001 

* This uses only V2 syntax. In the previous PR, I was looking at the docs on the `main` branch for the action which has unreleased syntax (namely the `any` keyword). For [stability purposes](https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/finding-and-customizing-actions#adding-an-action-to-your-workflow) it is useful to reference the version of the action we are using (as opposed to referencing the `main` branch).
* Fixes some of the globs. For example, drops the leading `./` in `./**/*.rb`.

### How will this affect user permissions?

N/A - this doesn't touch application code and should not impact permissions.

### How is this tested? (please write tests!) 💖💪

It seems that actions only using code on the `master`/`main` branch. I found that the easiest way to iterate on testing was to work off [my fork](https://github.com/tmr08c/casa/pulls) where I could commit straight to master 🤠 . I was also able to make PRs and not alert a bunch of watchers. 

I added a note in `labeler.yml` to suggest this workflow for future developers. 

### Screenshots please :)

All PRs are still up [here](https://github.com/tmr08c/casa/pulls)

* List of PRs

Screen Shot 2020-10-08 at 8 55 26 PM

* Proof I didn't manually add them 😈 

Screen Shot 2020-10-08 at 8 55 41 PM

* Successful action run 

Screen Shot 2020-10-08 at 8 55 53 PM

### Feelings gif (optional)
What gif best describes your feeling working on this issue? https://giphy.com/

just stay there labels!

_Me trying to throw labels at PRs_

### Next steps

If there are other labels we would like to add, please let me know - I can do it in this PR, or we can move forward with this and I can make future PRs.

### CODE REVIEW

File: .github/labeler.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+#
+# Warning - Testing changes to this file can be tricky
+# At the time of writing, you need to have changes committed to master for the
+# Action to use them. This isn't ideal for rapid iteration. If you want to make
+# changes to this file and experiment with changes, please consider testing the
+# changes in a fork of this repo first. This allows you to add changes to
+# master as needed and also create multiple Pull Requests to confirm labels are
+# properly added.
+
+ruby:
```
Comment (tmr08c) [inline .github/labeler.yml:13]: Key is label name, list is globs that we will check file changes against. If a changed file matches the glob, we add the label :)

File: .github/labeler.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# Configuration for the Labeler action
+# The action is set up in ./workflows/label.yml
+# See https://github.com/actions/labeler for more details
+#
+# Warning - Testing changes to this file can be tricky
```
Comment (tmr08c) [inline .github/labeler.yml:5]: Hopefully, this is useful advice for others and clear in the advice.

Another tip I learned is how to include debug messages in the action logs (assuming the action has debug logging): https://github.com/actions/toolkit/blob/main/docs/action-debugging.md#how-to-access-step-debug-logs :)

**Review by compwron — APPROVED** [review]
very very very cool!!!! ✨✨✨✨

**Comment by compwron** [comment]
I just want to say again that I am VERY VERY happy with these labels, thank you so much for getting this working!!! :)

**Comment by tmr08c** [comment]
@compwron really glad to hear this ❤️!

It looks like maybe there’s two slightly different versions of the test label (https://github.com/rubyforgood/casa/pull/1033). Not quite sure how that happened but hopefully as things get auto-labeled we can settle on one. 

Let me know if you think of anything that would be good to auto-label.

**Comment by compwron** [comment]
You are invited to the CASA asynchronous retrospective-and-futurespective! Please take a look at https://bit.ly/casaretro (no login required) and add at least one note! Even if it's just "I was here and you are cool" or "omg y'all why do you keep messaging me" we want to hear it. <3
Next week Wednesday 18 November @ 6pm Pacific time we will Zoom about the board and talk about themes and takeaways. (Zoom link in our calendar at http://bit.ly/casacal or slack #casa channel https://rubyforgood.herokuapp.com/ )
Please ESPECIALLY add lots of notes if you won't be able to make it on Wednesday to talk!
* Note: the board will be monitored for spam; message me or any team lead if you see a problem
https://bit.ly/casaretro

**Pull request merged**